obspy/io/seiscomp/event.py

```diff
@@ -20,43 +20,71 @@
+    Simple function checking if the passed object contains a valid sc3ml file.
+    Returns True of False.
+
+    The test is not exhaustive - it only checks the root tag but that should
+    be good enough for most real world use cases. If the schema is used to
+    test for a StationXML file, many real world files are false negatives as
+    they don't adhere to the standard.
+
+    :type path_or_file_object: str
+    :param path_or_file_object: File name or file like object.
+    :rtype: bool
+    :return: ``True`` if SC3ML file is valid.
+    """
+    return _is_sc3ml_version(path_or_file_object, SCHEMA_VERSION)
 
-    if isinstance(path_or_object, etree._Element):
-        xmldoc = path_or_object
-    else:
-        try:
-            xmldoc = etree.parse(path_or_object)
-        except etree.XMLSyntaxError:
-            if verbose:
-                print('Not an XML file')
-            return False
 
-    valid = xmlschema.validate(xmldoc)
+def _read_sc3ml(filename, id_prefix='smi:org.gfz-potsdam.de/geofon/'):
+    """
+    Read a 0.9 SC3ML file and returns a :class:`~obspy.core.event.Catalog`.
 
-    # Pretty error printing if the validation fails.
-    if verbose and valid is not True:
-        print("Error validating SC3ML file:")
-        for entry in xmlschema.error_log:
-            print("\t%s" % entry)
+    An XSLT file is used to convert the SC3ML file to a QuakeML file. The
+    catalog is then generated using the QuakeML module.
 
-    return valid
+    .. warning::
+    This function should NOT be called directly, it registers via the
+    the :meth:`~obspy.core.event.catalog.Catalog.write` method of an
+    ObsPy :class:`~obspy.core.event.catalog.Catalog` object, call this
+    instead.
+
+    :type filename: str
+    :param filename: SC3ML file to be read.
+    :type id_prefix: str
+    :param id_prefix: ID prefix. SC3ML does not enforce any particular ID
+        restriction, this ID prefix allows to convert the IDs to a well
+        formatted QuakeML ID. You can modify the default ID prefix with the
+        reverse DNS name of your institute.
+    :rtype: :class:`~obspy.core.event.Catalog`
+    :return: An ObsPy Catalog object.
+
+    .. rubric:: Example
+
+    >>> from obspy import read_events
+    >>> cat = read_events('/path/to/iris_events.sc3ml')
+    >>> print(cat)
+    2 Event(s) in Catalog:
+    2011-03-11T05:46:24.120000Z | +38.297, +142.373
+    2006-09-10T04:26:33.610000Z |  +9.614, +121.961
+    """
+    xslt_filename = os.path.join(os.path.dirname(__file__), 'data',
+                                 'sc3ml_0.9__quakeml_1.2.xsl')
+    transform = etree.XSLT(etree.parse(xslt_filename))
+    sc3ml_doc = _xml_doc_from_anything(filename)
+    quakeml_doc = transform(sc3ml_doc,
+                            ID_PREFIX=etree.XSLT.strparam(id_prefix))
+    return Unpickler().load(io.BytesIO(quakeml_doc))
 
 
 def _write_sc3ml(catalog, filename, validate=False, verbose=False,
```

Comment:
I don't see any `open()` in your quote. Moreover, the [lxml FAQ](http://lxml.de/FAQ.html#can-lxml-parse-from-file-objects-opened-in-unicode-text-mode) states that:

```text
if possible, prefer passing the file path directly into parse() instead of an opened Python file object.
```

Maybe you're speaking of [this](https://github.com/obspy/obspy/blob/master/obspy/io/seiscomp/event.py#L107) `open()`, which was copied from the [QuakeML module](https://github.com/obspy/obspy/blob/master/obspy/io/quakeml/core.py#L1832). I can eventually use a `with open()` but I'll also modifie the QuakeML module then.